File: Beginner/A - Books (BIT2A)/books.cpp
#include<iostream>
using namespace std;

int main() {
    int t;
    cin>>t;
    while(t--) {
        int N;
        cin>>N;
        int A[N], ans[N];

        for (int i = 0; i < N; i++)
            cin>>A[i];
        
        int s;
        for (int i = 0; i < N; i++) {
            s = 0;
            for (int j = i; j < N; j++) {
                if (A[j] == A[i])   
                    continue;
                else
                    s++;
            }
            ans[i] = s;
        }
        for (int i = 0; i < N; i++)
            cout<<ans[i]<<" ";
        cout<<endl;
    }
}

File: Beginner/A - Books (BIT2A)/books.go
package main

import (
	"fmt"
	"strconv"
	"strings"
)

func main() {
	var t int
  fmt.Scan(&t)
	for i := 0; i < t; i++{
		var n int
		fmt.Scan(&n)
		a := make([]int, n)
		b := make([]int, n)
		ans := make([]string, n)
		for j := 0; j < n; j++ {
			fmt.Scan(&a[j])
		}
		for j := 0; j < n; j++ {
			for k := 0; k < n; k++ {
				if a[k] > a[j] {
					b[j]++
				}
			}
		}
		for j := 0; j < n; j++ {
			ans[j] = strconv.Itoa(b[j])
		}
		fmt.Println(strings.Join(ans, " "))
	}
	return
}
